src/workerd/server/server.c++

```diff
@@ -1304,79 +1304,72 @@ private:
-
-      auto encodedName = kj::encodeUriComponent(cacheName);
+    kj::Own<kj::HttpClient>
+    getNamespace(kj::StringPtr cacheName, kj::Maybe<kj::String> cfBlobJson,
+                 kj::Maybe<Tracer::Span &> parentSpan) override {
 
-      return kj::heap<CacheHttpClientImpl>(
-          parent,  kj::mv(encodedName), kj::mv(cfBlobJson), parentSpan);
+      return kj::heap<CacheHttpClientImpl>(cacheService, kj::str(cacheName),
+                                           kj::mv(cfBlobJson), parentSpan);
     }
 
   private:
-    Service& parent;
+    Service &cacheService;
   };
 
-  class CacheHttpClientImpl final: public kj::HttpClient {
+  class CacheHttpClientImpl final : public kj::HttpClient {
   public:
-    CacheHttpClientImpl(Service& parent,
-                        kj::Maybe<kj::String> cacheName, kj::Maybe<kj::String> cfBlobJson,
-                        kj::Maybe<Tracer::Span&> parentSpan)
-        : cacheName(kj::mv(cacheName)),
-          cfBlobJson(kj::mv(cfBlobJson)), parentSpan(kj::mv(parentSpan)) {
-      client = asHttpClient(parent.startRequest({kj::mv(cfBlobJson), parentSpan}));
-    }
+    CacheHttpClientImpl(Service &parent, kj::Maybe<kj::String> cacheName,
+                        kj::Maybe<kj::String> cfBlobJson,
+                        kj::Maybe<Tracer::Span &> parentSpan)
+        : client(asHttpClient(
+              parent.startRequest({kj::mv(cfBlobJson), parentSpan}))),
+          cacheName(kj::mv(cacheName)) {}
 
     Request request(kj::HttpMethod method, kj::StringPtr url,
                     const kj::HttpHeaders &headers,
                     kj::Maybe<uint64_t> expectedBodySize = nullptr) override {
 
-      return client->request(method, url, addCacheNameHeader(headers, cacheName),
-                           expectedBodySize);
+      return client->request(method, url,
+                             addCacheNameHeader(headers, cacheName),
+                             expectedBodySize);
     }
 
   private:
     kj::Own<kj::HttpClient> client;
     kj::Maybe<kj::String> cacheName;
-    kj::Maybe<kj::String> cfBlobJson;
-    kj::Maybe<Tracer::Span&> parentSpan;
 
-    kj::HttpHeaders addCacheNameHeader(const kj::HttpHeaders& headers,
-       kj::Maybe<kj::StringPtr> cacheName) {
-      kj::ArrayPtr<const char> spliceParts[3];
-      uint spliceCount = 0;
+    kj::HttpHeaders addCacheNameHeader(const kj::HttpHeaders &headers,
+                                       kj::Maybe<kj::StringPtr> cacheName) {
       auto headersCopy = headers.cloneShallow();
-      KJ_IF_MAYBE(name, cacheName) {
-        spliceParts[spliceCount++] = "{\"namespace\":\""_kj;
-        spliceParts[spliceCount++] = *name;
-        spliceParts[spliceCount++] = "\"}"_kj;
-      } else {
-        spliceParts[spliceCount++] = "{}"_kj;
+      KJ_IF_MAYBE (name, cacheName) {
+        auto encodedName = kj::encodeUriComponent(*name);
+
+        headersCopy.set(IoContext::current().getHeaderIds().cfCacheNamespace,
```

Comment:
I don't think it's necessarily safe to assume that there is a current `IoContext` here. It would be better if the header ID could be passed down to `CacheClientImpl` via its constructor.

Reply:
To be clear, it shouldn't be necessary to use `IoContext::current()` at all... I think somewhere further up the stack you should be able to access the `ThreadContext` directly, or maybe the `HttpHeaderTable::Builder` directly.